### ISSUE TITLE
append_to_file does nothing if line matches what's already in destination file

### ISSUE THREAD

**Issue by mingchang** [piotrmurach/tty-file#4]
Trying to append a line (which happens to already exist in destination file) returns false, and skips appending altogether. This created a lot of confusion for me in instances such as:

```
        TTY::File.append_to_file(file) do
          "\n"
        end
```

Where "\n" already exists somewhere in 'file' and tty-file decided to skip appending. From debugger output it looks like we're jumping into 'inject_into_file' and subsequently 'replace_in_file' to do the actual appending, but 'replace_in_file' does a regex match and skips appending if found.

```
[556, 565] in /Users/ming/.rvm/gems/ruby-2.4.0/gems/tty-file-0.6.0/lib/tty/file.rb
   556: 
   557:       log_status(:replace, relative_path, options.fetch(:verbose, true),
   558:                                           options.fetch(:color, :green))
   559:       return false if options[:noop]
   560: 
=> 561:       if !(contents =~ /^#{match}(\r?\n)*/m) || options[:force]
   562:         status = contents.gsub!(*args, &block)
   563:         if !status.nil?
   564:           ::File.open(relative_path, 'wb') do |file|
   565:             file.write(contents)

```

I've been able to work around the issue by simply doing a 

```
        TTY::File.append_to_file(file, force: true) do
          "\n"
        end
```

but it took a bit of time in the debugger trying to figure out why append_to_file was actually skipping arbitrary lines that I was trying to append.

I'd like to suggest that append_to_file could default to a behavior similar to force: true, so that it skips checking whether the line already exists. Potentially append_to_file could be a forced append, while append_to_file(file, if_new: true) could follow the existing behavior. I think that could be a lot more intuitive for new users.

**Comment by piotrmurach** [comment]
Thanks for using this gem! ❤️ 

Apologies it took me so long to get around to answering this issue!

I had very similar discussion related to this in another issue https://github.com/piotrmurach/tty-file/issues/3

The reason why I implemented `append_to_file` to check first if the line being appended already exists is due to [teletype](https://github.com/piotrmurach/tty) generators. It was important to ensure 

I will see what I can do, I'm thinking of changing the option to be named `:safe`. So by default the operation would be `unsafe` and just append anythig but you could always ensure safety by passing `safe: true`.

**Comment by mingchang** [comment]
This is quite a good gem BTW, thanks for taking the time to make it 👍 .

I think even if the change could not be made, just a sentence in the readme saying "WARNING: append_to_file doesn't append if the text already exists, please be sure to use the force: true option if you would like to append unconditionally" could be very helpful.

I'm actually not too familiar with teletype and how it causes the append to need to be "safe", but to me the word "append" has not implication of involving any conditionality, so another idea is the append_to_file can be unsafe only, while another function append_with_check could be the safe option.

I'm not too picky exactly how it ends up being sorted out, but just wanted to point out that it was confusing to me when I used the function, and hopefully we can make it a little more intuitive.

**Comment by piotrmurach** [comment]
@mingchang I've changed it to always append, inject, prepend content without checking and added `safe_apped_to_file`, `safe_prepend_to_file`, `safe_inject_into_file` to provide alternative versions that better express the intention. Updated README to reflect the changes. Released in latest version `0.7.0`.

**Comment by mingchang** [comment]
@piotrmurach Amazing to hear. Thanks a ton :)
